File: algorithms/1909.remove-one-element-to-make-the-array-strictly-increasing/solution.go
func canBeIncreasing(nums []int) bool {
	pre := nums[0]
	removed := false

	for i := 1; i < len(nums); i++ {
		if pre < nums[i] {
			pre = nums[i]
		} else {
			if removed {
				return false
			}
			removed = true

			if i == 1 || nums[i-2] < nums[i] {
				pre = nums[i]
			}
		}
	}

	return true
}

File: algorithms/1909.remove-one-element-to-make-the-array-strictly-increasing/solution.rs
impl Solution {
    pub fn can_be_increasing(nums: Vec<i32>) -> bool {
        let mut pre = nums[0];
        let mut removed = false;

        for i in 1..nums.len() {
            if pre < nums[i] {
                pre = nums[i];
            } else {
                if removed {
                    return false;
                }
                removed = true;

                if i == 1 || nums[i - 2] < nums[i] {
                    pre = nums[i];
                }
            }
        }

        true
    }
}
